build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactmetrics/AnalyzeMetricsTask.kt

```diff
@@ -0,0 +1,148 @@
+                    )
+
+                    val remoteMetrics = remoteMetricsFile.toMap()
+                    val remoteLines = remoteMetricsFile.readLines()
+                    val remoteHeaders = remoteLines[0].split(",").map { it.trim() }
+
+                    val localMetrics = metricsFile.get().asFile.toMap()
+                    val localLines = metricsFile.get().asFile.readLines()
+                    val localHeaders = localLines[0].split(",").map { it.trim() }
+
+                    val artifactNames = remoteHeaders.toSet() + localHeaders.toSet()
+                    val metrics = mutableMapOf<String, DataPoint>()
+
+                    var addedArtifact = false
+                    var removedArtifact = false
+                    var significantChange = false
+
+                    artifactNames.forEach { artifact ->
+                        val local = localMetrics[artifact] ?: 0
+                        val remote = remoteMetrics[artifact] ?: 0
+
+                        val diff = local - remote
+                        val percentage = buildString {
+                            if (remote == 0L || local == 0L) {
+                                append("NaN")
+
+                                if (remote == 0L) addedArtifact = true
+                                if (local == 0L) removedArtifact = true
+
+                                return@buildString
+                            }
+
+                            if (diff < 0) append("-")
+
+                            val percentageValue = (abs(diff).toDouble() / remote.toDouble() * 100).roundToInt()
+                            if (percentageValue > 5 && diff > 0) significantChange = true
+
+                            append(percentageValue)
+                            append("%")
+                        }
+
+                        metrics[artifact] =
+                            DataPoint(
+                                remote,
+                                local,
+                                diff,
+                                percentage,
+                            )
+                    }
+
+                    shouldFailWorkflow.get().asFile.writeText(significantChange.toString())
+
+                    val diffTable = buildString {
+                        appendLine("Affected Artifacts\n=")
+                        appendLine("| Artifact |Pull Request (bytes) | Latest Release (bytes) | Delta (bytes / percentage) |")
+                        appendLine("| -------- | ------------------- | ---------------------- | -------------------------- |")
+                        metrics.forEach { artifact ->
+                            if (artifact.value.diff != 0L) {
+                                appendLine("|${artifact.key}|${"%,d".format(artifact.value.localSize)}|${"%,d".format(artifact.value.latestReleaseSize)}|${"%,d".format(artifact.value.diff)} / ${artifact.value.percentage}|")
+                            }
+                        }
```

Comment:
**Suggestion**: Consider splitting the delta column into two and then right-aligning each column except for the artifact names:

```md
| Artifact   | Pull Request (bytes) | Latest Release (bytes) | Delta (bytes) | Delta (%) |
|------------|---------------------:|-----------------------:|--------------:|----------:|
| aws-config | 500,000              | 500,000                | 0             | 0%        |
| aws-core   | 30,000               | 40,000                 | -10,000       | -25%      |
…
```

This will yield a table like:

| Artifact   | Pull Request (bytes) | Latest Release (bytes) | Delta (bytes) | Delta (%) |
|------------|---------------------:|-----------------------:|--------------:|----------:|
| aws-config | 500,000              | 500,000                | 0             | 0%        |
| aws-core   | 30,000               | 40,000                 | -10,000       | -25%      |